EJERCICIO_HERENCIA_PRODUCTOS_COCA.cpp

```diff
@@ -0,0 +1,219 @@
+class Ropa : public Producto {
+public:
+	float compra;
+	int cantidadRopa;
+public:
+	Ropa(string _nombreRopa, string _descripcionRopa, int _precio, int _cantidadRopa) : Producto() {
+		this->nombre = _nombreRopa;
+		this->descripcion = _descripcionRopa;
+		this->precio = _precio;
+
+		this->garantia = " 1 anio";
+		this->descuento = 0.1;
+		this->precioTotal = 0;
+		this->costoEnvio = 0;
+		this->cantidadRopa = _cantidadRopa;
+	}
+
+	float ObtenerDescuento() {
+		descuento = 0.1;
+		return descuento;
+	}
+	int costodeEnvio() {
+		costoEnvio = 0;
+		return costoEnvio;
+	}
+	float ObtenerDescuentoPrecioFinal() {
+		float tot;
+		tot = descuento + costoEnvio + precio;
+		return tot;
+	}
+};
+
+class Libro : public Producto {
+public:
+	float compra;
+	int cantidadLibros;
+public:
+	 //Opcion 3
+	Libro(string nombreLibro, string descripcionLibro, int precio, int _cantidadLibros) : Producto() {
+		this->nombre = nombreLibro;
+		this->descripcion = descripcionLibro;
+		this->precio = precio;
+
+		this->garantia = " 30 días";
+		this->descuento = 0.05;
+		this->precioTotal = 0;
+		this->costoEnvio = 0;
+		this->cantidadLibros = _cantidadLibros;
+	}
+	
+	float ObtenerDescuento() {
+		descuento = 0.1;
+		return descuento;
+	}
+	int costodeEnvio() {
+		costoEnvio = 0;
+		return costoEnvio;
+	}
+	float ObtenerDescuentoPrecioFinal() {
+		float tot;
+		tot = descuento + costoEnvio + precio;
+		return tot;
+	}
+};
+class ListaProductos {
+private:
+    Producto** listaProductos;
+    int cantidad;
+    int capacidad;
+
+public:
+    ListaProductos(int _capacidad) {
+        this->cantidad = 0;
+        this->capacidad = _capacidad;
+        this->listaProductos = new Producto * [_capacidad];
+    }
+
+    void agregarProducto(Producto* nuevoProducto) {
+        if (cantidad < capacidad) {
+            this->listaProductos[cantidad] = nuevoProducto;
+            cantidad++;
+        }
+    }
+
+    void mostrarProductos() {
+        for (int i = 0; i < cantidad; i++) {
+            this->listaProductos[i]->mostrarProducto();
+        }
+    }
+
+};
+int main()
+{
+// Declaración lista 
+    int cantidadMaxProductos = 10;
+    ListaProductos* productos = new ListaProductos(cantidadMaxProductos);
+
+    //Agregar productos
+	//_nombre, _descripcion, _precio, _garantia, _descuento, _precioTotal, _costoEnvio
+    Libro* libro1 = new Libro("Tilin", "Horror ", 10, 2);
+    productos->agregarProducto(libro1);
+	libro1->ObtenerDescuentoPrecioFinal();
+	libro1->mostrarProducto();
+	cout << "-----------" << endl;
+	Ropa* ropa1 = new Ropa("Blusa", "Rosa ", 100, 2);
+	productos->agregarProducto(ropa1);
+	ropa1->ObtenerDescuentoPrecioFinal();
+	ropa1->mostrarProducto();
+	cout << "-----------" << endl;
+	Electrodomesticos* elec1 = new Electrodomesticos("Lavadora", "Ultimo modelo ", 1000, 1);
+	productos->agregarProducto(elec1);
+	elec1->ObtenerDescuentoPrecioFinal();
+	elec1->mostrarProducto();
```

Comment:
Mantener consistente si se va a mantener Pascal o camel case
```suggestion
	productos->agregarProducto(elec1);
	elec1->obtenerDescuentoPrecioFinal();
	elec1->mostrarProducto();
```